Mathlib/Data/Finset/Basic.lean

```diff
@@ -3995,3 +3998,30 @@ end List
 -- Note that we cannot use `List.sublists` itself as that is defined very early.
 assert_not_exists List.sublistsLen
 assert_not_exists Multiset.powerset
+
+namespace Mathlib.Meta
+open Qq Lean Meta Finset
+
+/-- Attempt to prove that a finset is nonempty using the `finsetNonempty` aesop rule-set.
+
+You can add lemmas to the rule-set by tagging them with either:
+* `aesop safe apply (rule_sets [finsetNonempty])` if they are always a good idea to follow or
+* `aesop unsafe apply (rule_sets [finsetNonempty])` if they risk directing the search to a blind
+  alley.
+-/
+def proveFinsetNonempty {u : Level} {α : Q(Type u)} (s : Q(Finset $α)) :
+    MetaM (Option Q(Finset.Nonempty $s)) := do
+  -- Aesop expects to operate on goals, so we're going to make a new goal.
+  let goal ← Lean.Meta.mkFreshExprMVar q(Finset.Nonempty $s)
+  let mvar := goal.mvarId!
+  -- We want this to be fast, so use only the basic and `Finset.Nonempty` specific rules.
+  let rulesets ← Aesop.Frontend.getGlobalRuleSets #[`builtin, `Finset.Nonempty]
```

Comment:
I suppose you were right that the ruleset name and the predicate name were confusing! This was the ruleset name, not the predicate name:
```suggestion
  let rulesets ← Aesop.Frontend.getGlobalRuleSets #[`builtin, `finsetNonempty]
```

Reply:
To debug this, I did `set_option trace.Tactic.positivity.failure true` and saw:
```
[Tactic.positivity.failure] no such rule set: 'Finset.Nonempty'
      (Use 'declare_aesop_rule_set' to declare rule sets.
       Declared rule sets are not visible in the current file; they only become visible once you import the declaring file.)
```

Reply:
:facepalm: